test/gcs_signed_url/crypto_test.exs

```diff
@@ -16,4 +18,68 @@ defmodule GcsSignedUrl.CryptoTest do
       assert Fixtures.foo_signed_64() == MUT.sign64("foo", client)
     end
   end
+
+  describe "sign_via_api/2" do
+    setup do
+      [
+        oauth_config: %SignBlob.OAuthConfig{
+          access_token: "Some access token",
+          service_account: "Some service account"
+        },
+        string_to_sign: "string-to-sign"
+      ]
+    end
+
+    test "returns signature upon successful response from the HTTP client", %{
+      oauth_config: oauth_config,
+      string_to_sign: string_to_sign
+    } do
+      MockSetup.sign_via_api()
+      assert {:ok, "signature"} == MUT.sign_via_api(string_to_sign, oauth_config)
+    end
+
+    test "returns error with details upon 401 response from API", %{
```

Comment:
Feels like over testing all the status codes and all the errors. Another idea is to create an Errors module where you could have `def forbidden(message), do: {:error, ...}`, `def unauthorized(message), do: {:error, ...}`

Reply:
Partially agree. I want to make sure that the expected responses of the API return correct errors. But I can combine the several types of non-200 in a single test. Still, there's the case where the call fails completely that I would like to cover.